backend/core/configuration.py

```diff
@@ -512,6 +528,65 @@ def handle_hijack_learn_rule_request(self, message):
                     priority=4,
                 )
 
+        def handle_load_as_sets(self, message):
+            """
+            Receives a "load-as-sets" message, translates the corresponding
+            as anchors into lists, and rewrites the configuration
+            :param message:
+            :return:
+            """
+            with open(self.file, "r") as f:
+                raw = f.read()
+            yaml_conf = ruamel.yaml.load(
+                raw, Loader=ruamel.yaml.RoundTripLoader, preserve_quotes=True
+            )
+            load_made = False
+            error = False
+            if "asns" in yaml_conf:
+                for name in yaml_conf["asns"]:
+                    if translate_as_set(name, just_match=True):
```

Comment:
can't we call `translate_as_set` once here?

Reply:
what do you mean? Cache the result so that if other ocurrences of the same name happen in the asn section we do not re-retrieve it. This is a good idea, I will implement it (it is easy).

Reply:
Fixed, see updates. Now when the button is pressed, each ASN is resolved exactly once.